ir/function.cpp

```diff
@@ -568,6 +679,44 @@ void LoopAnalysis::analysis() {
       uf.merge(x, w);
     }
   }
+
+  if (type[0] != NodeType::nonheader)
```

Comment:
why this special case?

Reply:
If a CFG has loops, for getting the roots, we have two cases: A: the entry block is a loop, or B: the entry block is not a loop but some of its children are loops. Those two cases are exclusive (if A then no B and vice versa) and needs to be considered separately.

Reply:
thanks!